hack/lib/util.sh

```diff
@@ -729,8 +730,9 @@ function kube::util::ensure-cfssl {
 # Check if we have "docker buildx" commands available
 #
 function kube::util::ensure-docker-buildx {
+  DOCKER=${DOCKER:-docker}
   # podman returns 0 on `docker buildx version`, docker on `docker buildx`. One of them must succeed.
-  if docker buildx version >/dev/null 2>&1 || docker buildx >/dev/null 2>&1; then
+  if "${DOCKER}" buildx version >/dev/null 2>&1 || "${DOCKER}" buildx >/dev/null 2>&1; then
```

Comment:
In https://github.com/kubernetes-sigs/kind/pull/3394/files the buildx subcommand was dropped?

Reply:
I dropped it in that pull request for `podman` because it did not seem needed. But that PR has since been closed and I might file one that better aligns with this `DOCKER=...` approach.

Keeping it here and thus keeping the semantic changes to a minimum, as it shouldn't harm.

Reply:
Er, terse comment, I wanted to clarify that podman has buildx, which also would've made the other PR less controversial :-)

Some of our images in both repos leverage features that were introduced by buildx that are not in legacy `docker build`, and in some places simply cannot be accomplished with legacy docker build (e.g. setcap preservation)

Reply:
You didn't mention in the other PR that this part was controversial. ;-)

Reply:
I mean we outlined that we are targeting this specific build tool, if podman also ships it ...

Reply:
IIRC, podman does not ship it as a separate tool but implements mostly the same CLI.